lib/facility.rb

```diff
@@ -1,14 +1,75 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
-
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles
+  
+  def initialize(facility_data)
+    @name = facility_data[:name] 
+    @address = facility_data[:address] 
+    @phone = facility_data[:phone] 
     @services = []
+    @registered_vehicles = []
   end
-
-  def add_services(service)
+  
+  def add_service(service)
     @services << service
   end
+
+  def collected_fees
+    @collected_fees = 0
```

Comment:
Since this variable isn't used outside of the `collected_fees` method, it should be a local variable, not an ivar. 
A different approach would be to store `@collected_fees` as an attribute in the initialize, and only do a `@collected_fees +=` at the time of vehicle registration. Your solution works well, just wanted to mention an alternative since we're on the subject of instance variables vs local variables.